File: Sheet 7 (Arrays 2)/H.cpp
#include <iostream>
#include <algorithm>
using namespace std;

int main() {
    int n; cin >> n; int arr[12]; int sum = 0;
    for (int i = 0; i < 12; i++) cin >> arr[i];

    if (n == 0) {
        cout << 0 << endl;
        return 0;
    }

    sort(arr, arr + 12, greater<int>());

    for (int i = 0; i < 12; i++) {
        sum += arr[i];
        if (sum >= n) {
            cout << i + 1 << endl;
            return 0;
        }
    }

    cout << -1 << endl;
    return 0;
}

File: Sheet 7 (Arrays 2)/H.py
k = int(input())
lst = list(map(int, input().split()))
lst.sort()
lst.reverse()
sum, i = 0, 0
while i < len(lst) and sum < k:
    sum += lst[i]
    i += 1
if sum < k:
    i = -1
print(i)
